product.go

```diff
@@ -84,12 +93,106 @@ type ProductsResource struct {
 	Products []Product `json:"products"`
 }
 
+// Pagination of results
+type Pagination struct {
+	NextPageOptions     *ListOptions
+	PreviousPageOptions *ListOptions
+}
+
 // List products
 func (s *ProductServiceOp) List(options interface{}) ([]Product, error) {
+	products, _, err := s.ListWithPagination(options)
+	if err != nil {
+		return nil, err
+	}
+	return products, nil
+}
+
+// ListWithPagination lists products and return pagination to retrieve next/previous results.
+func (s *ProductServiceOp) ListWithPagination(options interface{}) ([]Product, *Pagination, error) {
 	path := fmt.Sprintf("%s.json", productsBasePath)
 	resource := new(ProductsResource)
-	err := s.client.Get(path, resource, options)
-	return resource.Products, err
+	headers := http.Header{}
+
+	headers, err := s.client.createAndDoGetHeaders("GET", path, nil, options, resource)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Extract pagination info from header
+	linkHeader := headers.Get("Link")
+
+	pagination, err := extractPagination(linkHeader)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return resource.Products, pagination, nil
+}
+
+// Extract pagination from the Link header
```

Comment:
Super minor point but the Go convention is for the godoc string to begin with the name of the element it describes, in this case rather than `Extract` `extractPagination`

https://blog.golang.org/godoc-documenting-go-code

Reply:
Thanks! Just fixed